gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/QueryAndGatewayDtoConverter.java

```diff
@@ -339,19 +340,16 @@ private static Function convertToQueryFunction(
             .setFunctionName(function.getFunction().name())
             .setAlias(function.getAlias());
 
-    // AVGRATE is adding a specific implementation because Pinot does not directly support this
-    // function
     switch (function.getFunction()) {
       case AVGRATE:
         {
-          builder.setFunctionName(FunctionType.SUM.name()).setAlias(function.getAlias());
-
-          // Adds only the argument that is a column identifier for now.
-          List<org.hypertrace.gateway.service.v1.common.Expression> columns =
-              function.getArgumentsList().stream()
-                  .filter(org.hypertrace.gateway.service.v1.common.Expression::hasColumnIdentifier)
-                  .collect(Collectors.toList());
-          columns.forEach(e -> builder.addArguments(convertToQueryExpression(e)));
+          builder.setFunctionName(function.getFunction().name()).setAlias(function.getAlias());
+          if (function.getArgumentsCount() > 0) {
+            function
+                .getArgumentsList()
+                .forEach(
+                    e -> builder.addArguments(convertToQueryExpression(convertLongToIsoFormat(e))));
```

Comment:
This is functionally correct, but as only literal argument we would like to convert to iso duration string, for nit (readable), can we do as below?
```
function.getArgumentsList().stream()
              .map(e -> e.hasLiteral() ? convertLongToIsoFormat(e) : e)
              .forEach(e -> builder.addArguments(convertToQueryExpression(e)));
```